content/automate/ManageIQ/Transformation/Ansible.class/__methods__/launchplaybookasaservice.rb

```diff
@@ -0,0 +1,51 @@
+module ManageIQ
+  module Automate
+    module Transformation
+      module Ansible
+        class LaunchPlaybookAsAService
+          def initialize(handle = $evm)
+            @handle = handle
+          end
+
+          def target_host(task, transformation_hook)
+            target_host = nil
+            case transformation_hook
+            when 'pre'
+              target_host = task.source
+            when 'post'
+              target_host = @handle.vmdb(:vm).find_by(:id => task.get_option(:destination_vm_id))
+            end
+            target_host
+          end
+
+          def main
+            task = @handle.root['service_template_transformation_plan_task']
+            transformation_hook = @handle.inputs['transformation_hook']
+
+            unless transformation_hook == '_'
+              service_template = task.send("#{transformation_hook}_ansible_playbook_service_template")
+              @handle.log(:info, "Service Template Id: #{service_template.id}")
+              unless service_template.nil?
+                target_host = target_host(task, transformation_hook)
+                unless target_host.nil?
```

Comment:
@fdupont-redhat 
Can this be a positive case with an 'if' statement instead of an unless, it makes it easier to read

```
 if service_template
.....
   target_host = target_host(task, transformation_hook)
   if target_host
       ......
   end
end
```
Also what happens if you can't find a target_host or service_template should we report on that?

Reply:
rubocop actually prefer earlier termination. In this example you can code this way
```
return if service_template.nil?
...
return if target_host.nil?
...
return unless target_host.power_state == 'on'
...